server/src/main/java/org/opensearch/cluster/decommission/DecommissionStatus.java

```diff
@@ -16,6 +16,11 @@ public enum DecommissionStatus {
      * Decommission process is initiated, and to-be-decommissioned leader is excluded from voting config
      */
     INIT("init"),
+    /**
+     * Decommission process is initiated, and the zone is being weighed away.
+     */
+    WEIGH_AWAY("weigh_away"),
+
```

Comment:
The life cycle of this new status should be - 
1. Setting the weights
2. Wait for it to drain
3. Handle failures of above two gracefully
4. Only on success of first two we should call node removal as is

We can rename this to `DRAINING` or `WEIGHING_AWAY`. I prefer first

Reply:
Updated the name to `DRAINING`